src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Transmission/OtlpExporterPersistentStorageRetryTransmissionHandler.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using Google.Protobuf;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+using OpenTelemetry.PersistentStorage.Abstractions;
+using OpenTelemetry.PersistentStorage.FileSystem;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Transmission;
+
+internal sealed class OtlpExporterPersistentStorageRetryTransmissionHandler<TRequest> : OtlpExporterTransmissionHandler<TRequest>
+{
+    private readonly ManualResetEvent stopEvent = new(false);
+    private readonly Thread thread;
+    private readonly PersistentBlobProvider? persistentBlobProvider;
```

Comment:
It looks like this is always set in the ctor so seems you could make this...

```csharp
private readonly PersistentBlobProvider? persistentBlobProvider;
```

Which would allow you to avoid the `null` checks below.